simulator-ui/src/main/java/org/citrusframework/simulator/ui/filter/CookieCsrfFilter.java

```diff
@@ -0,0 +1,28 @@
+package org.citrusframework.simulator.ui.filter;
+
+import jakarta.servlet.FilterChain;
+import jakarta.servlet.ServletException;
+import jakarta.servlet.http.HttpServletRequest;
+import jakarta.servlet.http.HttpServletResponse;
+import org.springframework.security.web.csrf.CsrfToken;
+import org.springframework.web.filter.OncePerRequestFilter;
+
+import java.io.IOException;
+
+/**
+ * Spring Security 6 doesn't set a XSRF-TOKEN cookie by default. This solution is
+ * <a href="https://github.com/spring-projects/spring-security/issues/12141#issuecomment-1321345077"> recommended by Spring Security.</a>
+ */
+public class CookieCsrfFilter extends OncePerRequestFilter {
```

Comment:
I think that this class is not required, but I must test this further:
- [ ] remove `CookieCsrfFilter` if it is unused (as guessed)